packages/babel-plugin-proposal-object-rest-spread/test/fixtures/object-rest/nested-default-value-2/output.js

```diff
@@ -0,0 +1,9 @@
+const {
+  InputProps: {
+    classes
+  } = {
+    foo: 1
+  }
+} = props,
+      InputPropsOther = babelHelpers.objectWithoutProperties(props.InputProps, ["classes"]),
```

Comment:
The default value should probably be declared as a new variable and then referenced here e.g. 
```suggestion
      InputPropsOther = babelHelpers.objectWithoutProperties(props.InputProps === undefined ? defaultInputProps : props.InputProps, ["classes"]),
```

Reply:
Thanks for the test case and possible solution! I will work on this case.

Note that for the `AssignmentPattern`, we should memorize the transformed member expression so that we would not execute them twice:
```js
const ref$propsInputProps = props.InputProps;
InputPropsOther = babelHelpers.objectWithoutProperties( 
  ref$propsInputProps === undefined ? defaultInputProps : ref$propsInputProps,
  ["classes"]
),
```

It is similar to what babel has done on array rest. ([REPL](https://babeljs.io/repl#?babili=false&browsers=chrome%2036&build=&builtIns=false&spec=false&loose=false&code_lz=G4QwTgBA3gUBEgFwQNp3hARuANO-AdEQPYAuAFgKZjoC6EAvNBAGbHHIAsATBAL54-jaHwDcQA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=&version=7.7.3&externalPlugins=))